src/Data/Show/Generic.purs

```diff
@@ -0,0 +1,53 @@
+module Data.Show.Generic
+  ( class GenericShow
+  , genericShow'
+  , genericShow
+  , class GenericShowArgs
+  , genericShowArgs
+  ) where
+
+import Prelude (class Show, show, (<>))
+import Data.Generic.Rep
+import Data.Symbol (class IsSymbol, reflectSymbol)
+import Type.Proxy (Proxy(..))
+
+class GenericShow a where
+  genericShow' :: a -> String
+
+class GenericShowArgs a where
+  genericShowArgs :: a -> Array String
+
+instance genericShowNoConstructors :: GenericShow NoConstructors where
+  genericShow' a = genericShow' a
+
+instance genericShowArgsNoArguments :: GenericShowArgs NoArguments where
+  genericShowArgs _ = []
+
+instance genericShowSum :: (GenericShow a, GenericShow b) => GenericShow (Sum a b) where
+  genericShow' (Inl a) = genericShow' a
+  genericShow' (Inr b) = genericShow' b
+
+instance genericShowArgsProduct
+    :: (GenericShowArgs a, GenericShowArgs b)
+    => GenericShowArgs (Product a b) where
+  genericShowArgs (Product a b) = genericShowArgs a <> genericShowArgs b
+
+instance genericShowConstructor
+  :: (GenericShowArgs a, IsSymbol name)
+  => GenericShow (Constructor name a) where
+  genericShow' (Constructor a) =
+      case genericShowArgs a of
+        [] -> ctor
+        args -> "(" <> intercalateImpl " " ([ctor] <> args) <> ")"
+    where
+      ctor :: String
+      ctor = reflectSymbol (Proxy :: Proxy name)
+
+instance genericShowArgsArgument :: Show a => GenericShowArgs (Argument a) where
+  genericShowArgs (Argument a) = [show a]
+
+-- | A `Generic` implementation of the `show` member from the `Show` type class.
+genericShow :: forall a rep. Generic a rep => GenericShow rep => a -> String
+genericShow x = genericShow' (from x)
+
+foreign import intercalateImpl :: String -> Array String -> String
```

Comment:
Don’t we usually drop the Impl suffix from foreign values which are used as-is instead of being used to implement PureScript values?

Reply:
I don't know. Do we have a documented list of conventions used across core libs?

I'll change the name, but I'd like to understand how to include prior git history from the generics-rep repo first before I make it.